app/assets/javascripts/dashboard.js

```diff
@@ -499,73 +504,51 @@ function selectedMastersLength() {
   return $('input[name="roles[master][]"]:checked').length;
 }
 
+
 function isBootstrappable() {
+  var errors = [];
+
   // We need at least one master
   if (selectedMastersLength() < 1) {
-    return false;
+    errors.push("You haven't selected one master at least");
+  }
+
+  // We need at least one worker
+  if (selectedWorkersLength() < 1) {
+    errors.push("You haven't selected one worker at least");
   }
 
   // We need an odd number of masters
   if (selectedMastersLength() % 2 !== 1) {
-    return false;
+    errors.push('The number of masters has to be an odd number');
   }
 
-  // We need at least one worker
-  if (selectedWorkersLength() < 1) {
-    return false;
-  }
+  State.bootstrapErrors = errors;
 
-  return true;
+  return errors.length === 0;
 }
 
 // return true if the selected masters vs workers are in a supported state
 function isSupportedConfiguration() {
-  // bootstrappable but may not be in a supported state
-  // (e.g.: 1 master, 1 workers)
-  if (!isBootstrappable()) {
-    return false;
-  }
-
   // We need at least three nodes in total
-  if ((selectedWorkersLength() + selectedMastersLength()) < 3) {
-    return false;
-  }
-
-  return true;
+  return (selectedWorkersLength() + selectedMastersLength()) >= 3;
```

Comment:
Can't we remove `isSupportedConfiguration`? I think we could just use `isBootstrappable` with an option to update `State.bootstrapErrors` only when we clicked on `Next`. I think it would simplify some logic, since `isSupportedConfiguration` is just a simplified version of `isBootstrappable`, no?

Reply:
We need `isSupportedConfiguration` to decide if we show or not the warning modal when 1 master and 1 worker and also the minimum nodes alert.

Reply:
Fair enough.